erts/emulator/beam/bif.tab

```diff
@@ -623,7 +623,7 @@ bif io:printable_range/0
 bif re:inspect/2
 
 ubif erlang:is_map/1
-gcbif erlang:map_size/1
+ubif erlang:map_size/1
```

Comment:
Does that call the `map_size_1` function in `erl_map.c`? If so, it calls `HAlloc` and that's supposedly not allowed now, isn't it?

Reply:
Good catch.

Yes, calling `HAlloc()` is not allowed and if `hsz` (the size argument passed to `HAlloc()` would ever be nonzero, the runtime system would crash.

In practice, it is not possible to create a map whose size can't be represented in a small. So I have pushed a commit that removes the call to `HAlloc()` and the superfluous calls to `erts_bld_uint()`.

Reply:
Would it be possible to set some kind of debug flag that triggers an assert if HAlloc is called in an incorrect function?

Reply:
I have added assertions that htop and stop are not changed. That should catch any calls to `HAlloc()` that allocate anything. Having a separate `HAllocForbidden` flag could potentially catch calls `HAlloct()` that don't actually allocate anything, but is seems to be overkill.